src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -419,6 +418,203 @@ DWORD WINAPI CProfilerManager::InstrumentationMethodThreadProc(
+        return 1;
+    }
+#endif
+
+    CProfilerManager* pProfilerManager = static_cast<CProfilerManager*>(lpParameter);
+    if (pProfilerManager == NULL)
+    {
+        CLogging::LogError(_T("CConfigurationLocator::ParseAttachConfigurationThreadProc - Invalid parameter"));
+        return 1;
+    }
+
+    CComPtr<CXmlDocWrapper> pDocument;
+    pDocument.Attach(new CXmlDocWrapper());
+    IfFailRet(pDocument->LoadContent(pProfilerManager->m_wszConfigXml));
+
+    CComPtr<CXmlNode> pDocumentNode;
+    IfFailRet(pDocument->GetRootNode(&pDocumentNode));
+
+    CComPtr<CXmlNode> pCurrChildNode;
+    IfFailRet(pDocumentNode->GetChildNode(&pCurrChildNode));
+
+    CComBSTR bstrCurrChildNodeName;
+    IfFailRet(pDocumentNode->GetName(&bstrCurrChildNodeName));
+
+    if (wcscmp(bstrCurrChildNodeName, _T("InstrumentationEngineConfiguration")) != 0)
+    {
+        CLogging::LogError(_T("CConfigurationLocator::ParseAttachConfigurationThreadProc - Invalid configuration. Root element should be InstrumentationEngineConfiguration"));
+        return E_FAIL;
+    }
+
+    while (pCurrChildNode != nullptr)
+    {
+        IfFailRet(pCurrChildNode->GetName(&bstrCurrChildNodeName));
+        if (wcscmp(bstrCurrChildNodeName, _T("InstrumentationEngine")) == 0)
+        {
+            CComPtr<CXmlNode> pSettingsNode;
+            IfFailRet(pCurrChildNode->GetChildNode(&pSettingsNode));
+
+            CComBSTR bstrSettingsNodeName;
+            IfFailRet(pSettingsNode->GetName(&bstrSettingsNodeName));
+
+            if (wcscmp(bstrSettingsNodeName, _T("Settings")) == 0)
+            {
+                unordered_map<tstring, tstring> settingsMap;
+                IfFailRet(ParseSettingsConfigurationNode(pSettingsNode, settingsMap));
+
+                for (unordered_map<tstring, tstring>::iterator it = settingsMap.begin();
+                     it != settingsMap.end();
+                     it++)
+                {
+                    if (wcscmp(it->first.c_str(), _T("LogLevel")) == 0)
+                    {
+                        CLogging::SetLoggingFlags(CLoggerService::ExtractLoggingFlags(it->second.c_str()));
+                    }
+                    else if (wcscmp(it->first.c_str(), _T("LogFileLevel")) == 0)
+                    {
+                        ::SetEnvironmentVariable(_T("MicrosoftInstrumentationEngine_FileLog"), it->second.c_str());
+                    }
+                    else if (wcscmp(it->first.c_str(), _T("LogFilePath")) == 0)
+                    {
+                        ::SetEnvironmentVariable(_T("MicrosoftInstrumentationEngine_FileLogPath"), it->second.c_str());
+                    }
+                    
+                }
+            }
+        }
+        else if (wcscmp(bstrCurrChildNodeName, _T("InstrumentationMethods")) == 0)
+        {
+            CComPtr<CXmlNode> pInstrumentationMethodNode;
+            IfFailRet(pCurrChildNode->GetChildNode(&pInstrumentationMethodNode));
```

Comment:
>GetChildNode [](start = 38, length = 12)

Can you check if GetChildNode returns an error if there are no child nodes? If so, we don't want to fail the parsing, but just don't try to process on the nonexistent node.

Reply:
GetChildNode will return a nullptr, good point on catching this

---
In reply to: [386705329](https://github.com/microsoft/CLRInstrumentationEngine/pull/216#discussion_r386705329) [](ancestors = 386705329)